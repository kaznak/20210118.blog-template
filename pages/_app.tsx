import type { AppProps } from 'next/app'
import Head from 'next/head'

import '@fontsource/roboto/300.css'
import '@fontsource/roboto/400.css'
import '@fontsource/roboto/500.css'
import '@fontsource/roboto/700.css'

import { ThemeProvider, useTheme } from '@material-ui/core/styles'
import CssBaseline from '@material-ui/core/CssBaseline'
import { mUiTheme } from '../lib/materialUiUtils'

function MyApp({ Component, pageProps }: AppProps) {
  return (
    <ThemeProvider theme={mUiTheme}>
      <Head>
        {/* PWA primary color */}
        <meta name="theme-color" content={useTheme().palette.primary.main} />
      </Head>
      <CssBaseline />
      <Component {...pageProps} />
    </ThemeProvider>
  )
}

export default MyApp
