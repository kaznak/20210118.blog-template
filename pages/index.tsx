import Typography from '@material-ui/core/Typography'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import Link from '../component/atoms/Link'

import { posts } from '../lib/mdxUtils'

export function Index({ posts }) {
  return (
    <>
      <Typography variant="h1">Home Page</Typography>
      <Typography>
        Click the link below to navigate to a page generated by{' '}
        <code>next-mdx-remote</code>.
      </Typography>
      <List>
        {posts.map((post) => (
          <ListItem key={post.slug}>
            <Link as={`/posts/${post.slug}`} href={`/posts/[slug]`}>
              {post.data.title}
            </Link>
          </ListItem>
        ))}
      </List>
    </>
  )
}

export function getStaticProps() {
  return { props: { posts } }
}

export default Index
