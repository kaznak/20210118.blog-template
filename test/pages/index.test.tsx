import React from 'react'
import { render } from '../testUtils'
import { Index, getStaticProps } from '../../pages'

const { props } = getStaticProps()

describe('Home page', () => {
  it('matches snapshot', () => {

    const { asFragment } = render(<Index {...props} />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
