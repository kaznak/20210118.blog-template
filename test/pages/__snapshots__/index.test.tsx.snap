// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    Home Page
  </h1>
  <p>
    Click the link below to navigate to a page generated by 
    <code>
      next-mdx-remote
    </code>
    .
  </p>
  <ul>
    <li>
      <a
        href="/posts/example-post"
      >
        Example Post
      </a>
    </li>
    <li>
      <a
        href="/posts/hello-world"
      >
        Hello World
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
